### PR TITLE
Liquibase support, book soft delete and search (by author, price) were added

### PR DESCRIPTION
**Pull request by MaxSkory**

### CODE REVIEW

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Please, fix the conflicts
Знімок екрана 2023-10-11 о 22 22 34

File: src/main/java/mskory/bookstore/repository/book/specs/AuthorSpecificationProvider.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.Arrays;
+import mskory.bookstore.model.Book;
+import mskory.bookstore.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+
```
Comment (kozhukhovsky) [inline src/main/java/mskory/bookstore/repository/book/specs/AuthorSpecificationProvider.java:11]:
```suggestion
```

File: src/main/resources/db/changelog/changes/02-add-is-deleted-column-books-table.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      changes:
+        - addColumn:
+            tableName: books
+            columns:
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValue: "false"
+                  constraints:
+                    nullable: false
```
Comment (kozhukhovsky) [inline src/main/resources/db/changelog/changes/02-add-is-deleted-column-books-table.yaml:14]:
```suggestion
                    nullable: false
                    
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+databaseChangeLog:
+  - include: 
+      file: db/changelog/changes/01-create-books-table.yaml
+  - include:
+      file: db/changelog/changes/02-add-is-deleted-column-books-table.yaml
+  - include:
+      file: db/changelog/changes/03-add-scale-to-price-column-book-table.yaml
```
Comment (kozhukhovsky) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: db/changelog/changes/03-add-scale-to-price-column-book-table.yaml
      
```

**Review by kozhukhovsky — APPROVED** [review]

**Comment by MaxSkory** [comment]
sorry, forgot to merge previous PR 😕

**Pull request merged**